chrometool/chrometool_test.go

```diff
@@ -0,0 +1,317 @@
+package chrometool
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/chromedp/chromedp"
+	"github.com/gobwas/ws"
+	"github.com/gobwas/ws/wsutil"
+)
+
+type requestType struct {
+	method string
+	params string
+}
+
+type responseType struct {
+	index        int
+	responseList [][]byte
+}
+
+var (
+	testServerPort = flag.Int("port", 8600, "port")
+	mockChromePort = ":0"
+	WsURL          = "ws://127.0.0.1" + mockChromePort
```

Comment:
Make this a flag, so that a user can actually run against a real chrome browser without having to make code changes.
Set default value to "", and override it with your mockChrome returned port thingy, *only* if it is a empty string (if user specified it, don't change it).